lapis2-docs/src/components/ConfigGenerator/Results.tsx

```diff
@@ -0,0 +1,86 @@
+import { type FC, useContext, useMemo } from 'react';
+import { ConfigContext } from './configContext';
+import { dump } from 'js-yaml';
+import { pathoplexusConfigSchema } from './pathoplexusConfig.ts';
+import { siloConfigSchema } from './siloConfig.ts';
+
+export const Results: FC = () => {
+    const { config, configType } = useContext(ConfigContext);
+
+    let zodParseResult;
+    let configToExport: object;
+    if (configType === 'Pathoplexus') {
+        configToExport = {
+            instanceName: config.instanceName,
+            opennessLevel: config.opennessLevel,
+            metadata: config.metadata,
+            website: {
+                tableColumns: config.tableColumns,
+            },
+            silo: {
+                dateToSortBy: config.dateToSortBy,
+                partitionBy: config.partitionBy,
+            },
+        };
+        zodParseResult = pathoplexusConfigSchema.safeParse(configToExport);
+    } else {
+        configToExport = config;
+        zodParseResult = siloConfigSchema.safeParse(configToExport);
+    }
+
+    if (zodParseResult.success) {
+        configToExport = zodParseResult.data;
+    }
+
+    const yaml = useMemo(() => dump({ schema: configToExport }), [configToExport]);
+
+    function handleDownload() {
+        const blob = new Blob([yaml], { type: 'text/plain' });
+        const url = URL.createObjectURL(blob);
+
+        const a = document.createElement('a');
+        a.href = url;
+        a.download = `config.yaml`;
+        a.click();
+
+        URL.revokeObjectURL(url);
+    }
+
+    const tooltipMessage = 'Please check that you have provided a primary key.';
```

Comment:
What can actually go wrong here? Should we indicate a bit better why you can't download yet?

I can think of at least one more error case: the instance name is not set.

Reply:
I added a ticket for this and updated the description.